core/templates/components/state-editor/state-skill-editor/state-skill-editor.component.ts

```diff
@@ -55,16 +59,35 @@ export class StateSkillEditorComponent implements OnInit {
     private windowDimensionsService: WindowDimensionsService,
     private stateLinkedSkillIdService: StateLinkedSkillIdService,
     private urlInterpolationService: UrlInterpolationService,
-    private ngbModal: NgbModal
+    private ngbModal: NgbModal,
+    private skillBackendApiService: SkillBackendApiService,
+    private userService: UserService
   ) {}
 
   ngOnInit(): void {
     this.skillEditorIsShown = (!this.windowDimensionsService.isWindowNarrow());
-    this.topicsAndSkillsDashboardBackendApiService.fetchDashboardDataAsync()
-      .then((response: TopicsAndSkillDashboardData) => {
-        this.categorizedSkills = response.categorizedSkillsDict;
-        this.untriagedSkillSummaries = response.untriagedSkillSummaries;
+    this.userService.getUserInfoAsync()
+      .then((userInfo) => {
+        this.isEditableByUser = (
+          userInfo.isLoggedIn() &&
+          (userInfo.isCurriculumAdmin() || userInfo.isTopicManager())
+        );
```

Comment:
Based on the changes, it looks like the skill editor will show empty results for users who do not have access to the topics and skills dashboard. I think the expected behaviour should be that anyone who visits the editor page for a published lesson should be able to see the skill editor in read-only mode (this is how other editor components work e.g. hints, interaction editor). Only those with sufficient rights will be allowed to update the skills.

Reply:
Updated the videos and modified the code. Now, any user can access the skill-editor in read only mode and people with rights can update the skills.